### PR TITLE
[Project Addition] Fracture Detection using DL

### PR DESCRIPTION
**Pull request by Arihant-Bhandari**
# Pull Request for DL-Simplified 💡

## Issue Title : #751 Fracture Detection using DL 

- **Info about the related issue (Aim of the project)**: Classifying images of bones into fractured and not-fractured.
- **Name:** Arihant Bhandari
- **GitHub ID:** https://github.com/Arihant-Bhandari
- **Email ID:** arihant.bhandari.2004@gmail.com
- **Idenitfy yourself:** GSSoC 2024 Contributor

Closes: #751 Fracture Detection using DL

### Describe the add-ons or changes you've made 📃

Wrote code for preprocessing and loading images from bone_fracture_binary_classification directory, turned labels into probability distributions, checked for image appropriateness, created models including custom CNN as well as pre-trained models, like ResNet-50, VGG16 and so on.

## Type of change ☑️

What sort of change have you made:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested? ⚙️

| Models | Validation Accuracy | Validation Loss | Test Accuracy | Test Loss |
|--------|---------------------|--------------------------| ---------------------|--------------------------|
| ResNet-50 | 38.74% | 9.8734 | 44.92% | 8.8777 |
| InceptionV3 | 61.26%  | 6.1766 | 55.08% | 7.1615 |
| CNN | 99.87% | 0.0100 | 100.00% | 0.0037 |
| VGG16 | 94.63% | 0.1482 | 96.16% | 0.1142 |
| MobileNet | 99.21% | 0.0346 | 100.00% | 0.0156 |
| DenseNet-121 | 94.50% | 0.1413 | 95.03% | 0.1494 |
| Xception | 98.82% | 0.0541 | 100.00% | 0.0170 |

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by Arihant-Bhandari** [comment]
hi @abhisheks008 this is my work on fracture detection, please review this PR, thank you as always for your time and patience; if any changes will gladly do things over.

**Comment by abhisheks008** [comment]
I'll review your PR by EOD.

**Comment by Arihant-Bhandari** [comment]
also @abhisheks008 i want your opinion on something, the other project i had proposed about cat species, i am getting 88% approx on the best model currently, would it be considered sufficient for apporval ? or should i try getting it to 90 and beyond ?

**Comment by abhisheks008** [comment]
> also @abhisheks008 i want your opinion on something, the other project i had proposed about cat species, i am getting 88% approx on the best model currently, would it be considered sufficient for approval ? or should i try getting it to 90 and beyond ?

Try to reach to the 90 mark

**Review by abhisheks008 — APPROVED** [review]
Approved @Arihant-Bhandari
